### PR TITLE
ARM FPU rework

### PR DESCRIPTION
**Pull request by Deryugin**
* Add support for VFP9-S
* Add test for FPU
* Rework ARM/NEON operations
* Support of FPv4/v5 for ARMv7-M. That is, stm32f7 and stm32f4 discovery now support FPU.

### CODE REVIEW

File: src/arch/arm/armlib/exceptions/entry.S [hunk 1]
```diff
@@ -32,62 +32,68 @@ irq_handler:
...
+
+	ldmfd  r0, {r0, r1} /* tmp reg */
+	stmfd sp!, {lr}
+
 	/*regs & SPSR on theirs places, as before handler */
 	/*returning from interrupt handler, CPSR <- SPSR */
 	ldmfd  sp!, {pc}^
 	/*continue from place, where interrupt raised */
 
+
```
Comment (Deryugin) [inline src/arch/arm/armlib/exceptions/entry.S]: Remove clear lines, please

File: src/arch/arm/fpu/vfp9_s.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+#include <embox/unit.h>
+#include <framework/mod/options.h>
+
+EMBOX_UNIT_INIT(vfp9_s_init);
+
+static int vfp9_s_init(void) {
+	uint32_t val;
+	uint32_t sid;
+
```
Comment (Deryugin) [inline src/arch/arm/fpu/vfp9_s.c]: Remove clear line, please.

File: src/arch/arm/fpu/vfp9_s.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/**
+ * @file
+ *
+ * @date May 24, 2018
+ * @author Anton Bondarev
+ */
+
+#ifndef SRC_ARCH_ARM_FPU_VFP9_S_H_
+#define SRC_ARCH_ARM_FPU_VFP9_S_H_
+
```
Comment (Deryugin) [inline src/arch/arm/fpu/vfp9_s.h]: Too much clear lines in this file as well. Remove it, please

File: src/compat/libc/math/Mybuild [hunk 4]
```diff
@@ -6,6 +6,7 @@ abstract module math {
 
 static module math_stub extends embox.compat.libc.math {
 	source "stub.c"
+	source "stub.h"
```
Comment (Deryugin) [inline src/compat/libc/math/Mybuild]: Is "stub.h" included anywhere?

File: templates/arm/qemu/mods.config [hunk 5]
```diff
@@ -9,14 +9,15 @@ configuration conf {
...
+	@Runlevel(0) include embox.arch.arm.fpu.vfp9_s(log_level=3)
 
 	@Runlevel(0) include embox.arch.arm.mmu_small_page(domain_access=1,v5_format=1)
 	@Runlevel(0) include embox.mem.vmem_alloc(pgd_align=0x4000,pmd_align=0x1000,pte_align=0x1000)
 	@Runlevel(0) include embox.mem.vmem
 
 	include embox.lib.debug.whereami
 
-	@Runlevel(1) include embox.driver.video.pl110
+	//@Runlevel(1) include embox.driver.video.pl110
```
Comment (alexkalmuk) [inline templates/arm/qemu/mods.config]: Why is it required?

File: src/arch/arm/armlib/exceptions/entry.S [hunk 6]
```diff
@@ -32,62 +32,68 @@ irq_handler:
...
-
 	/* now we have previous mode and sp */
-	ARM_FPU_CONTEXT_SAVE_DEC(r1, sp)
-	stmfd  sp!, {r2-r12, lr}
 
 	/* restore spsr, lr, tmp reg */
 	ldmfd  r0!, {r1} /* spsr */
-	ldmfd  r0!, {r2} /* lr */
-	ldmfd  r0!, {r3, r4} /* tmp reg */
+    stmfd  sp!, {r1} /* spsr */
```
Comment (Deryugin) [inline src/arch/arm/armlib/exceptions/entry.S]: Fix tabs, please.

File: src/arch/arm/armlib/exceptions/exceptions.S [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+.global	undef_handler
+.global	prefetch_abt_handler
+.global	fiq_handler
+.global	prefetch_abt_handler
+.global arm_unresolvable_exception
+.global data_abt_handler
+
+undef_handler:
+#ifdef __ARM_NEON__
+    sub r14, r14, #4
```
Comment (Deryugin) [inline src/arch/arm/armlib/exceptions/exceptions.S]: Replace spaces with tabs, please.

File: src/drivers/interrupt/cortexm_irq_handle.S [hunk 8]
```diff
@@ -13,50 +13,62 @@
 interrupt_handle_enter:
 
     mov    r0, sp
-    # It should be a size of struct context, but we align it up to multiple of 8
-    # due to ARMv7-M manual (see AAPCS 5.2.1.2)
+    # "In the Cortex-M7 processor CCR.STKALIGN is read-only and has a value of 1.
+    # This means that the exception stack frame starting address
+    # is always 8-byte aligned."
+	# So we align it for 8 byte for all cortex-m
```
Comment (Deryugin) [inline src/drivers/interrupt/cortexm_irq_handle.S]: Replace tab with spaces, please

File: templates/arm/qemu/mods.config [hunk 9]
```diff
@@ -169,7 +170,13 @@ configuration conf {
 	@Runlevel(2) include embox.framework.LibFramework
 	@Runlevel(2) include embox.compat.libc.all
 	include embox.compat.libc.math_openlibm
+	//include embox.compat.libc.math_stub
```
Comment (Deryugin) [inline templates/arm/qemu/mods.config]: Remove it? arm/qemu has HW float now

**Comment by Deryugin** [comment]
LGTM

**Pull request merged**